hi-ml-multimodal/src/health_multimodal/image/model/model.py

```diff
@@ -75,7 +77,23 @@ class ImageModelOutput():
     projected_patch_embeddings: torch.Tensor
 
 
-class ImageModel(nn.Module):
+class BaseImageModel(nn.Module, ABC):
+    """ An abstract class defining methods required by the ImageInferenceEngine """
+
+    def __init__(self, **kwargs: Any):
+        super().__init__()
+        assert hasattr(self, "training")
```

Comment:
Why is this necessary?

Reply:
I suspect this `__init__` method can just be removed completely, but I might be wrong.

Reply:
I put it in initially because we use the `training` attribute in the inference engine, but I now realise we can assume it has one by virtue of being a `nn.module`, so lets remove this `__init__`!